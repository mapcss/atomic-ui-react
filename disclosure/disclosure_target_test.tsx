import DisclosureTarget from "./disclosure_target.ts";
import Disclosure from "./disclosure.ts";
import SSRProvider from "../ssr/ssr_provider.ts";
import { createElement, Fragment } from "react";
import {
  assertSnapshot,
  describe,
  expect,
  it,
  setupJSDOM,
} from "../dev_deps.ts";
import { render } from "@testing-library/react";

const describeTests = describe({
  name: "DisclosureTarget",
  async beforeEach() {
    await setupJSDOM();
  },
});

it(describeTests, "should throw error when it not wrap Context", () => {
  expect(() => render(<DisclosureTarget>test</DisclosureTarget>)).toThrow();
});

it(
  describeTests,
  "should render with style when the internal `isOpen` is false",
  (t) => {
    const { baseElement } = render(
      <DisclosureTarget>test</DisclosureTarget>,
      {
        wrapper: ({ children }) => {
          return (
            <SSRProvider>
              <Disclosure>{children as never}</Disclosure>
            </SSRProvider>
          );
        },
      },
    );

    assertSnapshot(t, baseElement.innerHTML);
  },
);

it(describeTests, "should render when the internal `isOpen` is true", (t) => {
  const { baseElement } = render(
    <DisclosureTarget>test</DisclosureTarget>,
    {
      wrapper: ({ children }) => {
        return (
          <SSRProvider>
            <Disclosure isDefaultOpen>{children as never}</Disclosure>
          </SSRProvider>
        );
      },
    },
  );

  assertSnapshot(t, baseElement.innerHTML);
});

it(describeTests, "should overwrite default closed style", (t) => {
  const { baseElement } = render(
    <DisclosureTarget
      closedStyle={{
        visibility: "hidden",
      }}
    >
      test
    </DisclosureTarget>,
    {
      wrapper: ({ children }) => {
        return (
          <SSRProvider>
            <Disclosure>{children as never}</Disclosure>
          </SSRProvider>
        );
      },
    },
  );

  assertSnapshot(t, baseElement.innerHTML);
});

it(describeTests, "should overwrite render", (t) => {
  const { baseElement } = render(
    <DisclosureTarget
      render={(props, { as, isOpen }) =>
        isOpen ? createElement(as, props) : createElement(Fragment)}
    >
      test
    </DisclosureTarget>,
    {
      wrapper: ({ children }) => {
        return (
          <SSRProvider>
            <Disclosure>{children as never}</Disclosure>
          </SSRProvider>
        );
      },
    },
  );

  assertSnapshot(t, baseElement.innerHTML);
});
