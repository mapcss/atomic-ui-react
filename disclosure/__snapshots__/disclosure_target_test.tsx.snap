export const snapshot = {};

snapshot[`DisclosureTarget > should render with style when the internal \`isOpen\` is false 1`] = `'<div><div display="none">test</div></div>'`;

snapshot[`DisclosureTarget > should render when the internal \`isOpen\` is true 1`] = `"<div><div>test</div></div>"`;

snapshot[`DisclosureTarget > should overwrite default closed style 1`] = `'<div><div visibility="hidden">test</div></div>'`;

snapshot[`DisclosureTarget > should overwrite render 1`] = `"<div></div>"`;
